### ISSUE TITLE
#[proc_macro_attribute] field injection is not supported by syntax highlighter

### ISSUE THREAD

**Issue by zb3qk** [intellij-rust/intellij-rust#9973]
## Environment

* **IntelliJ Rust plugin version:** 0.4.185.5086-223
* **Rust toolchain version:** 1.63.0 (4b91a6ea7 2022-08-08) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2022.3.1 (CL-223.8214.51)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled

## Problem description
`#[proc_macro_attribute]` does not inform syntax highlighting correctly in this [proc macro example for struct field injection](https://github.com/eonm-abes/proc-macro-issue-minimal-example). Opening this same files with `rust-analyzer` in VSCode does not have this issue.

## Steps to reproduce
1. Open the above Github project in CLion with the Rust IntelliJ plugin activated
2. Turn on Cargo check (on the fly calculation)
3. Navigate to tests/01-shoud-add-struct-field.rs
4. Observe that field `a` is linted with the error message `No such field`
5. ![image](https://user-images.githubusercontent.com/36089408/212391940-af6db393-4365-46fb-b0b1-f8a18db6b2d0.png)
6. Navigate to tests/02-parse-attributes.rs
7. Observe that field `a` is linted with the error message `No such field`
8. Update  `#[add_field2("a")]` to  `#[add_field2("b")]`
9. Observe a new linting error stating 
```
struct `tests::add_field::Foo` has no field named `a` [E0560] Help: a field with a similar name exists
```
10. ![image](https://user-images.githubusercontent.com/36089408/212392215-954943e2-c0f8-4797-9645-c0737f170ad1.png)

All code to reproduce this issue can be found in this github repository: https://github.com/eonm-abes/proc-macro-issue-minimal-example

**Comment by vlad20012** [comment]
Try to enable `org.rust.macros.proc.attr` experimental feature via the corresponding dialog. To open the dialog, just invoke `Help | Find Action` action and type `Experimental features`

**Comment by zb3qk** [comment]
Thanks for the pointer! I just set the experimental feature now:

![image](https://user-images.githubusercontent.com/36089408/212393282-c28ea745-fb6d-4d39-b079-f1a1a83e83fb.png)

It didn't seem to work just by activating the setting (there didn't seem to be a `apply` button or any restart dialog). I went ahead and invalidated the cache and reopened the IDE. This seemed to work! 

![image](https://user-images.githubusercontent.com/36089408/212393745-5908b7cc-a25e-4b2a-be8a-442566d43294.png)

Thanks for the help!
